apm-commons/apm-util/src/main/java/org/apache/skywalking/apm/util/ConfigInitializer.java

```diff
@@ -85,9 +85,16 @@ private static void initNextLevel(Properties properties, Class<?> recentConfigTy
                     } else {
                         // Convert the value into real type
                         final Length lengthDefine = field.getAnnotation(Length.class);
-                        if (lengthDefine != null && propertyValue.length() > lengthDefine.value()) {
-                            StringUtil.cut(propertyValue, lengthDefine.value());
-                            System.err.printf("The config value will be truncated , because the length max than %d : %s -> %s%n", lengthDefine.value(), configKey, propertyValue);
+                        if (lengthDefine != null) {
+                            int lengthLimited = lengthDefine.value();
+                            String lengthKey = String.format("%s#length", configKey);
+                            if (properties.containsKey(lengthKey)) {
+                                lengthLimited = Math.max(lengthLimited, Integer.valueOf(properties.getProperty(lengthKey, "0")));
+                            }
```

Comment:
I think `max` would be a little tricky. As we provided a configuration item, I would prefer to use the configuration value to override static codes as always. 
Could you explain why do you code as `max`?

Reply:
The default value is 50, which means it is the suggested value. If the configuration is smaller than it which means most of the service names or instance names will be cut especially in k8s environment.  

So we suggest the configuration for the length is better bigger than the default value.

Reply:
You are adding a feature to provide customization, I can't see why bigger is acceptable, but smaller isn't.
Your explanation seems doesn't make sense.

All length relative configurations are accepting value length in both ways. 

Please update the codes to provide an override capability, rather than within this condition.

Reply:
ok, I will change code to prefer user's configuration.

Reply:
@wu-sheng  done